mucommander-viewer-text/src/main/java/com/mucommander/viewer/text/TextViewer.java

```diff
@@ -185,7 +186,7 @@ void startEditing(AbstractFile file, DocumentListener documentListener) throws I
         }
     }
 
-    void loadDocument(InputStream in, final String encoding, DocumentListener documentListener) throws IOException {
+    void loadDocument(String fileExtension, InputStream in, final String encoding, DocumentListener documentListener) throws IOException {
```

Comment:
can't we get it from `currentFile` instead?

Reply:
we can, but I think we need to decide how to work out mime-type first, then we will update the code to fit the idea

Reply:
employed hint about FileTypeUtil#guessContentType